aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -597,6 +549,29 @@ void geometric_kernel_cuda(TensorIterator& iter, double p_, Generator* gen_) {
    });
 }
 
+void bernoulli_scalar_cuda_kernel(TensorIterator& iter, double p_, Generator* gen_) {
+  auto gen = check_generator<CUDAGenerator>(gen_, &globalContext().defaultGenerator(kCUDA));
+  AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Half, iter.dtype(), "bernoulli_scalar_cuda_", [&] {
+    using accscalar_t = at::acc_type<scalar_t, true>;
+    auto p = static_cast<accscalar_t>(p_);
+    // define lambda for bernoulli transformation
+    auto bernoulli_func = [p] __device__ (accscalar_t rand) {
+      return static_cast<scalar_t>(rand <= p);
+    };
+    if (std::is_same<scalar_t, double>::value) {
+      distribution_nullary_kernel<scalar_t, accscalar_t, curand4_engine_calls/2>(iter,
+        gen,
+        [] __device__ (curandStatePhilox4_32_10_t* state) { return curand_uniform2_double(state); },
```

Comment:
so @ngimel told me that curand's uniform with double is actually a lie because there is not enough precision. I don't know if things have changed or not.

Reply:
How many bits of randomness does it provide? 53 bits is standard so that it generates all the rationals `x / 2^53` for x in `[0,1, ..., 2^53-1]`.

That doesn't generate all valid doubles in [0, 1), but basically nobody does that outside of toy programs.

Reply:
`curand_uniform_double` is a lie (uses 32 bits), `curand_uniform2_double` is not (uses 53 bits per value).